DevContainer: build failing on main

*Title*: *Envoy build fails when using devcontainer mode on main branch*

*Description*:

```
vscode@docker-desktop:/workspaces/envoy$ bazel build -c dbg //source/exe:envoy-static
Starting local Bazel server and connecting to it...
INFO: Analyzed target //source/exe:envoy-static (1035 packages loaded, 46500 targets configured).
INFO: Found 1 target...
INFO: From Action external/opencensus_proto/opencensus/proto/agent/trace/v1/trace_service.grpc.pb.h:
bazel-out/k8-dbg/bin/external/opencensus_proto/external/opencensus_proto: warning: directory does not exist.
INFO: From Action external/com_github_grpc_grpc/src/proto/grpc/reflection/v1alpha/reflection.grpc.pb.h:
bazel-out/k8-dbg/bin/external/com_github_grpc_grpc/external/com_github_grpc_grpc: warning: directory does not exist.
INFO: From Action external/com_google_googleapis/google/devtools/cloudtrace/v2/trace.grpc.pb.h:
bazel-out/k8-dbg/bin/external/com_google_googleapis/external/com_google_googleapis: warning: directory does not exist.
ERROR: /build/.cache/bazel/_bazel_vscode/2d35de14639eaad1ac7060a4dd7e3351/external/com_google_cel_cpp/eval/public/BUILD:662:11: Compiling eval/public/value_export_util.cc failed: (Exit 1): clang-14 failed: error executing command (from target @com_google_cel_cpp//eval/public:value_export_util) /opt/llvm/bin/clang-14 -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer -g ... (remaining 92 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_google_cel_cpp/eval/public/value_export_util.cc:1:
In file included from external/com_google_cel_cpp/eval/public/value_export_util.h:7:
In file included from external/com_google_cel_cpp/eval/public/cel_value.h:37:
external/com_google_cel_cpp/base/memory.h:92:7: error: no matching function for call to 'operator delete'
      ::operator delete(static_cast<void*>(p), n * sizeof(T),
      ^~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/new:155:6: note: candidate function not viable: no known conversion from 'unsigned long' to 'std::align_val_t' for 2nd argument
void operator delete(void*, std::align_val_t, const std::nothrow_t&)
     ^
/usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/new:153:6: note: candidate function not viable: requires 2 arguments, but 3 were provided
void operator delete(void*, std::align_val_t)
     ^
/usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/new:144:6: note: candidate function not viable: requires 2 arguments, but 3 were provided
void operator delete(void*, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
     ^
/usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/new:180:13: note: candidate function not viable: requires 2 arguments, but 3 were provided
inline void operator delete  (void*, void*) _GLIBCXX_USE_NOEXCEPT { }
            ^
/usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/new:130:6: note: candidate function not viable: requires 1 argument, but 3 were provided
void operator delete(void*) _GLIBCXX_USE_NOEXCEPT
     ^
1 error generated.
Target //source/exe:envoy-static failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 1037.427s, Critical Path: 145.00s
INFO: 2116 processes: 1088 internal, 1028 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
```

Same here:
```bash
envoy on  main [$] via 🐹 v1.18.1 took 5s 
➜ bazel build //contrib/exe:envoy-static
WARNING: The following configs were expanded more than once: [clang]. For repeatable flags, repeats are counted twice and may lead to unexpected behavior.
WARNING: The following configs were expanded more than once: [clang]. For repeatable flags, repeats are counted twice and may lead to unexpected behavior.
INFO: Analyzed target //contrib/exe:envoy-static (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
ERROR: /data/fabrizio/ongres/github.com/envoy/source/extensions/access_loggers/filters/cel/BUILD:11:19: Compiling source/extensions/access_loggers/filters/cel/cel.cc failed: (Exit 1): clang failed: error executing command (from target //source/extensions/access_loggers/filters/cel:cel_lib) /usr/lib/llvm-14/bin/clang -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer ... (remaining 245 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from source/extensions/access_loggers/filters/cel/cel.cc:1:
In file included from ./source/extensions/access_loggers/filters/cel/cel.h:12:
In file included from ./source/extensions/filters/common/expr/evaluator.h:7:
In file included from ./source/extensions/filters/common/expr/context.h:11:
In file included from external/com_google_cel_cpp/eval/public/cel_value.h:37:
external/com_google_cel_cpp/base/memory.h:92:7: error: no matching function for call to 'operator delete'
      ::operator delete(static_cast<void*>(p), n * sizeof(T),
      ^~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:155:6: note: candidate function not viable: no known conversion from 'unsigned long' to 'std::align_val_t' for 2nd argument
void operator delete(void*, std::align_val_t, const std::nothrow_t&)
     ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:153:6: note: candidate function not viable: requires 2 arguments, but 3 were provided
void operator delete(void*, std::align_val_t)
     ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:144:6: note: candidate function not viable: requires 2 arguments, but 3 were provided
void operator delete(void*, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
     ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:180:13: note: candidate function not viable: requires 2 arguments, but 3 were provided
inline void operator delete  (void*, void*) _GLIBCXX_USE_NOEXCEPT { }
            ^
/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/12/new:130:6: note: candidate function not viable: requires 1 argument, but 3 were provided
void operator delete(void*) _GLIBCXX_USE_NOEXCEPT
     ^
1 error generated.
Target //contrib/exe:envoy-static failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 17.305s, Critical Path: 16.61s
INFO: 25 processes: 17 internal, 8 linux-sandbox.
FAILED: Build did NOT complete successfully
```

--- a/.devcontainer/Dockerfile
+++ b/.devcontainer/Dockerfile
@@ -5,7 +5,7 @@ ARG USER_UID=501
 ARG USER_GID=$USER_UID
 
 ENV BUILD_DIR=/build
-ENV ENVOY_STDLIB=libstdc++
+ENV ENVOY_STDLIB=libc++
 
 ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get -y update \

--- a/.devcontainer/setup.sh
+++ b/.devcontainer/setup.sh
@@ -5,7 +5,7 @@ BAZELRC_FILE=~/.bazelrc bazel/setup_clang.sh /opt/llvm
 # TODO(phlax): use user.bazelrc
 # Use generated toolchain config because we know the base container is the one we're using in RBE.
 # Not using libc++ here because clangd will raise some tidy issue in libc++ header as of version 9.
-echo "build --config=rbe-toolchain-clang" >> ~/.bazelrc
+echo "build --config=rbe-toolchain-clang-libc++" >> ~/.bazelrc
 echo "build ${BAZEL_BUILD_EXTRA_OPTIONS}" | tee -a ~/.bazelrc
 
 # Ideally we want this line so bazel doesn't pollute things outside of the devcontainer, but some of

